### PR TITLE
Docs addresses

### LINKED ISSUE #2011: Don't include arbitrary addresses in documentation.
One example is https://web3py.readthedocs.io/en/stable/web3.eth.html#web3.eth.Eth.send_raw_transaction. `0xd3CdA913deB6f67967B99D67aCDFa1712C293601` should be replaced with `public_address_of_receivers_account`.

This is live address that is actually used on mainnet: https://etherscan.io/address/0xd3CdA913deB6f67967B99D67aCDFa1712C293601 . Whether this is accident - I don't know.

### PR DESCRIPTION
**Pull request by reedsa**
### What was wrong?

Closes #2011 

### How was it fixed?

Replaced addresses in examples that may have risk involved, especially in transactions. Most of the time though, the addresses point to a real (but empty/unused) account.

### Todo:
- [X] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/main/newsfragments/README.md)

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: docs/overview.rst [hunk 1]
```diff
@@ -322,7 +322,8 @@ ENS
 ~~~
 
 `Ethereum Name Service (ENS) <https://ens.domains/>`_ provides the infrastructure
-for human-readable addresses. As an example, instead of
-``0xfB6916095ca1df60bB79Ce92cE3Ea74c37c5d359``, you can send funds to
-``ethereumfoundation.eth``. web3.py has support for ENS, documented
+for human-readable addresses. If an address is registered with the ENS registry,
+the domain name can be used in place of the address itself. For example, the registered domain 
+name ``ethereum.eth`` will resolve to the address
+``0xde0B295669a9FD93d5F28D9Ec85E40f4cb697BAe``. web3.py has support for ENS, documented
```
Comment (reedsa) [inline docs/overview.rst:328]: Updated this to get away from the "sending funds" recommendation 😬

File: docs/web3.eth.rst [hunk 2]
```diff
@@ -153,7 +153,7 @@ The following properties are available on the ``web3.eth`` namespace.
     .. code-block:: python
 
         >>> web3.eth.accounts
-        ['0xd3CdA913deB6f67967B99D67aCDFa1712C293601']
+        ['public_address']
```
Comment (wolovim) [inline docs/web3.eth.rst]: should this be the dummy 0x582 address?

**Review by wolovim — APPROVED** [review]
lgtm. 'jasoncarver.eth' still appears in some examples, which might be fine, but we could also consider swapping that out for something publicly maintained by ENS or EF. if desirable, fine as a separate PR too.

**Comment by reedsa** [comment]
Will track replacement of jasoncarver.eth in https://github.com/ethereum/web3.py/issues/3020

**Pull request merged**